theano/sandbox/cuda/opt.py

```diff
@@ -2399,6 +2399,34 @@ def local_gpu_extract_diagonal(node):
     return False
 
 
+@register_opt()
+@local_optimizer([theano.tensor.Diagonal])
```

Comment:
Why don't you make a single optimization with ExtractDiag's one? Am I missing something?